ros/processor/suggestions_engine.py

```diff
@@ -174,23 +176,32 @@ def download_and_extract(self, archive_URL, host, org_id):
         except Exception as error:
             logging.error(f"{self.service} - {self.event} - Error occurred during download and extraction: {error}")
 
-    def is_pcp_collected(self, platform_metadata):
+    def get_ros_pcp_status(self, platform_metadata):
         return (
-            platform_metadata.get('is_ros_v2') and
-            platform_metadata.get('is_pcp_raw_data_collected')
+            platform_metadata.get('is_ros_v2', False),
+            platform_metadata.get('is_pcp_raw_data_collected', False)
         )
 
     def handle_create_update(self, payload):
         self.event = "Update event" if payload.get('type') == 'updated' else "Create event"
 
         platform_metadata = payload.get('platform_metadata')
         host = payload.get('host')
+        is_ros_enabled, is_pcp_collected = self.get_ros_pcp_status(platform_metadata)
 
         if platform_metadata is None or host is None:
             logging.info(f"{self.service} - {self.event} - Missing host or/and platform_metadata field(s).")
             return
 
-        if not self.is_pcp_collected(platform_metadata):
+        if not is_ros_enabled:
+            return
+
+        if not is_pcp_collected:
+            logging.debug(
+                f"{self.service} - {self.event} - Triggering an event for system {host.get('id')}"
+            )
+            self.consumer.commit()
+            produce_report_processor_event(payload, platform_metadata, self.producer)
```

Comment:
The `payload` does contain the `platform_metadata`? 
We do have data contract, so in that context should we only send the fields which make sense to consume by our db schema? This will reduce the size of the message and avoid sending data which is not useful?

Reply:
That means we need to create a specific payload for all the cases we handle where `is_ros_v2` is `true` and `is_pcp_raw_collected` can be `true` or `false`, and events from the API. 
While not complicating the Kafka message schema, the data contract should be considered while working with the fields to be saved in the DB, specifically under report_processor.